blocks/carousel/carousel.js

```diff
@@ -0,0 +1,159 @@
+    if (el.id === targetPicture) {
+      el.classList.replace('hide', 'show');
+    } else {
+      el.classList.replace('show', 'hide');
+    }
+  });
+};
+
+const bottomNavigation = (event) => {
+  const target = event.target.id;
+  activateSlide(target);
+  activateBullet(target);
+};
+
+const arrowNavigation = (event) => {
+  const navButton = event.currentTarget.id;
+  let nextSlide = 0;
+  if (navButton === 'arrow-left') {
+    nextSlide = (activeSlide - 1) > 0 ? activeSlide - 1 : slideCount;
+  } else if (navButton === 'arrow-right') {
+    nextSlide = (activeSlide + 1) <= slideCount ? activeSlide + 1 : 1;
+  }
+  activateSlide(`slide-${nextSlide}`);
+  activateBullet(`slide-${nextSlide}`);
+  activeSlide = nextSlide;
+};
+
+const bottomNavOnClickEvents = () => {
+  const btnNav = document.querySelector('.bottom-nav');
+  [...btnNav.children].forEach((el) => {
+    el.addEventListener('click', bottomNavigation);
+  });
+};
+
+const arrowNavOnClickEvents = () => {
+  const arrowNav = document.querySelector('.arrow-nav');
+  [...arrowNav.children].forEach((el) => {
+    el.addEventListener('click', arrowNavigation);
+  });
+};
+
+// Control Slide show
+const toNextSlide = () => {
+  const nextSlide = (activeSlide + 1) <= slideCount ? activeSlide + 1 : 1;
+
+  activateSlide(`slide-${nextSlide}`);
+  activateBullet(`slide-${nextSlide}`);
+  activeSlide = nextSlide;
+};
+
+// Start slide show
+const startSlideShow = () => {
+  if (!slideShow) {
+    slideShow = window.setInterval(toNextSlide, 5000);
+  }
+};
+
+const stopSlideShow = () => {
+  slideShow = window.clearInterval(slideShow);
+};
+
+const initSlider = () => {
+  const sliderWrapper = document.querySelector('.slider-wrapper');
+  sliderWrapper.addEventListener('mouseover', stopSlideShow);
+  sliderWrapper.addEventListener('mouseleave', startSlideShow);
+  window.addEventListener('load', startSlideShow);
+};
+
+export default function decorate(block) {
+  // Set slide-wrapper class
+  const sliderWrapper = block.firstElementChild.lastElementChild;
+  sliderWrapper.classList.add('slider-wrapper');
+
+  // put pictures under slider flex container
+  const pictures = sliderWrapper.getElementsByTagName('picture');
+  const slider = document.createElement('div');
+  slider.classList.add('slider');
+  [...pictures].forEach((el) => {
+    slider.appendChild(el);
+  });
+  sliderWrapper.innerHTML = '';
+  sliderWrapper.appendChild(slider);
+
+  // set picture id slide 1 - n
+  let i = 1;
+  const slides = [];
+  [...slider.children].forEach((el) => {
+    el.setAttribute('id', `slide-${i}`);
+    el.classList.add(i === 1 ? 'show' : 'hide');
+    slides.push(`slide-${i}`);
+    i += 1;
+  });
+  slideCount = slides.length;
+
+  // Create Arrow nav
+  const arrowNavContainer = document.createElement('div');
+  arrowNavContainer.classList.add('arrow-nav');
+
+  const arrowLeft = document.createElement('a');
+  arrowLeft.setAttribute('id', 'arrow-left');
```

Comment:
the IDs should be a bit more specific, e.g. `slider-arrow-left` to avoid potential collisions with other features having arrows.